packages/headless/src/controllers/commerce/core/facets/date/headless-commerce-date-facet.ts

```diff
@@ -16,11 +17,18 @@ export type DateFacetOptions = Omit<
   'toggleSelectActionCreator' | 'toggleExcludeActionCreator'
 >;
 
+export type DateFacetState = CoreCommerceFacetState<DateFacetValue>;
+
 /**
  * The `DateFacet` controller offers a high-level programming interface for implementing date commerce
  * facet UI component.
  */
-export type DateFacet = CoreCommerceFacet<DateRangeRequest, DateFacetValue>;
+export type DateFacet = Omit<
+  CoreCommerceFacet<DateRangeRequest, DateFacetValue>,
```

Comment:
Instead of overriding the `state` prop manually in all facet types with `{ state: XFacetState }`, we could pass `CoreCommerceFacet` a third generic parameter defined as `FacetState extends CoreCommerceFacetState<ValueResponse>` with a value of `XFacetState`.

Reply:
Excellent idea!